src/libcollections/btree/map.rs

```diff
@@ -689,61 +689,69 @@ impl<K: Ord, V> BTreeMap<K, V> {
     {
         let min = range.start();
         let max = range.end();
-        let front = match min {
-            Included(key) => {
-                match search::search_tree(self.root.as_ref(), key) {
-                    Found(kv_handle) => {
-                        match kv_handle.left_edge().force() {
-                            Leaf(bottom) => bottom,
-                            Internal(internal) => last_leaf_edge(internal.descend()),
-                        }
+        let mut min_found = false;
+        let mut max_found = false;
+        let mut min_edge;
+        let mut max_edge;
+        let mut min_node = self.root.as_ref();
+        let mut max_node = self.root.as_ref();
+        let mut diverged = false;
+
+        loop {
+
+            min_edge = match (min_found, min) {
+                (false, Included(key)) => {
+                    match search::search_linear(&min_node, key) {
```

Comment:
search_tree searches the entire tree and returns a handle anywhere in the tree.  I need to search a single node at a time and be able to compare 2 searches.  You might think that I want search_node, but that returns a handle.  I need to be able to compare 2 edges of the same node, which I can't.  search_linear just gives me an index value into the node, which is exactly what I need.